test-data/unit/check-based-misc.test

```diff
@@ -21,3 +21,38 @@ if True:
     1  # type: ignore[operator]
 [out]
 main:3:5:3:31: error: Unused "type: ignore" comment  [unused-ignore]
+
+
+[case testFunctionAnnotations]
+# flags: --python-version=3.8
+def f():
+    a: list[int]
+    class A:
+        a: list[int]  # E: "list" is not subscriptable, use "typing.List" instead  [misc]
+        def f(self):
+            a: list[int]
+    def f(a: list[int]):  # E: "list" is not subscriptable, use "typing.List" instead  [misc]
+        ...
+
+[case testCastAtRuntime]
+# flags: --python-version=3.8
+from typing import cast
+cast(list[int], [])  # E: "list" is not subscriptable, use "typing.List" instead  [misc]
+cast("list[int]", [])
+
+
+[case testCastAtRuntimeFuture]
+# flags: --python-version=3.8
+from __future__ import annotations
+from typing import cast
+cast(list[int], [])  # E: "list" is not subscriptable, use "typing.List" instead  [misc]
+cast("list[int]", [])
+[builtins fixtures/tuple.pyi]
+
+
+[case testBasedTypesInFunction]
+def f():
+    a: 1 | 2
+    b: int & str
+    c: (int, str)
```

Comment:
do these also need to be tested in function signatures and casts, like in the other tests

Reply:
They do have some tests in their suits, I don't think we need heaps of tests for this, the logic is simple.